quadriga/src/main/java/edu/asu/diging/quadriga/core/db/impl/AppDbConnection.java

```diff
@@ -0,0 +1,74 @@
+package edu.asu.diging.quadriga.core.db.impl;
+
+import java.util.Random;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Component;
+
+import edu.asu.diging.quadriga.core.db.IAppDbConnection;
+import edu.asu.diging.quadriga.core.exceptions.UnstorableObjectException;
+import edu.asu.diging.quadriga.core.model.IApp;
+import edu.asu.diging.quadriga.core.model.impl.App;
+
+@Component
+@Transactional
+public class AppDbConnection implements IAppDbConnection {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public IApp getById(String id) {
+        return em.find(App.class, id);
+    }
+
+    @Override
+    public IApp store(IApp app) throws UnstorableObjectException {
+        if (app.getId() == null) {
+            throw new UnstorableObjectException("App does not have an id.");
+        } else {
+            em.persist(app);
+        }
+        em.flush();
+        return app;
+    }
+
+    @Override
+    public String generateId() {
+        String id = null;
+        while (true) {
+            id = "APP" + generateUniqueId();
+            Object existingFile = getById(id);
+            if (existingFile == null) {
+                break;
+            }
+        }
+        return id;
```

Comment:
check citesphere or virtual spaces  for how to create ids in this form (through annotation and a generator)